rocAL/rocAL/source/api/rocal_api_meta_data.cpp

```diff
@@ -231,8 +231,16 @@ ROCAL_API_CALL rocalGetImageLabels(RocalContext p_context, void* buf, unsigned i
     if(context->user_batch_size() != meta_data_batch_size)
         THROW("meta data batch size is wrong " + TOSTR(meta_data_batch_size) + " != "+ TOSTR(context->user_batch_size() ))
 
-    if (context->affinity == RocalAffinity::CPU)
-        memcpy(buf, meta_data.second->get_label_batch().data(), sizeof(int) * meta_data_batch_size);
+    if (context->affinity == RocalAffinity::CPU) {
```

Comment:
Not sure why this is changed from tot version. We need to keep all 3 cases (CPU only, GPU_HIP and GPU_OpenCL). Please revert back to tot

Reply:
This change was added since we added a memcpy_to_device parameter in pipeline which if set, passes a positive value to the flags variable in this function. We are checking for that value and if its set, assumes the user is asking to copy to a HIP buffer hence added an else block with hipMemcpy. I will add enable_hip and enable_opencl flags inside the else block to support GPU_HIP and GPU_OCL

Reply:
Added check for HIP/OCL backend and copying labels to buffers accordingly